File: refactor.js
const fs = require('fs');
const path = require('path');

const directory = 'lib/features';

const replacements = [
    [/const Color\(0xFFFFE0B2\)/g, 'AppColors.warningContainer'],
    [/const Color\(0xFFECF5FE\)/g, 'AppColors.surfaceContainerLow'],
    [/Color\(0x0D002349\)/g, 'AppColors.primaryContainer.withValues(alpha: 0.05)'],
    
    [/Radius\.circular\(\s*4(?:\.0)?\s*\)/g, 'AppRadii.sm'],
    [/Radius\.circular\(\s*8(?:\.0)?\s*\)/g, 'AppRadii.defaultRadius'],
    [/Radius\.circular\(\s*12(?:\.0)?\s*\)/g, 'AppRadii.md'],
    [/Radius\.circular\(\s*16(?:\.0)?\s*\)/g, 'AppRadii.lg'],
    [/Radius\.circular\(\s*24(?:\.0)?\s*\)/g, 'AppRadii.xl'],
    
    [/BorderRadius\.circular\(\s*4(?:\.0)?\s*\)/g, 'AppRadii.borderSm'],
    [/BorderRadius\.circular\(\s*8(?:\.0)?\s*\)/g, 'AppRadii.borderDefault'],
    [/BorderRadius\.circular\(\s*12(?:\.0)?\s*\)/g, 'AppRadii.borderMd'],
    [/BorderRadius\.circular\(\s*16(?:\.0)?\s*\)/g, 'AppRadii.borderLg'],
    [/BorderRadius\.circular\(\s*24(?:\.0)?\s*\)/g, 'AppRadii.borderXl'],
    
    [/(?<!Pdf)Colors\.red/g, 'AppColors.error'],
    [/(?<!Pdf)Colors\.green/g, 'AppColors.success'],
    [/(?<!Pdf)Colors\.amber/g, 'AppColors.warning'],
    [/(?<!Pdf)Colors\.grey\.shade400/g, 'AppColors.outlineVariant'],
    [/(?<!Pdf)Colors\.grey\.shade600/g, 'AppColors.outline'],
    [/(?<!Pdf)Colors\.grey\.shade100/g, 'AppColors.surfaceContainerHigh'],
    [/(?<!Pdf)Colors\.black26/g, 'AppColors.onSurface.withValues(alpha: 0.26)'],
    [/(?<!Pdf)Colors\.black54/g, 'AppColors.onSurface.withValues(alpha: 0.54)'],
    [/(?<!Pdf)Colors\.black\s*\.withValues\(\s*alpha:\s*0\.4\s*\)/g, 'AppColors.onSurface.withValues(alpha: 0.4)'],
    [/(?<!Pdf)Colors\.black\s*\.withValues\(\s*alpha:\s*0\.15\s*\)/g, 'AppColors.onSurface.withValues(alpha: 0.15)']
];

function walkDir(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach((file) => {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        if (stat && stat.isDirectory()) {
            results = results.concat(walkDir(fullPath));
        } else if (file.endsWith('.dart')) {
            results.push(fullPath);
        }
    });
    return results;
}

const files = walkDir(directory);
files.forEach(filepath => {
    let content = fs.readFileSync(filepath, 'utf8');
    let newContent = content;
    
    replacements.forEach(([regex, repl]) => {
        newContent = newContent.replace(regex, repl);
    });
    
    if (newContent !== content) {
        // Fix imports
        const normalizedPath = filepath.replace(/\\/g, '/');
        const depth = normalizedPath.split('/').length - 1;
        // lib/features/... -> depth is at least 2. (lib, features, file = depth 2)
        // lib/features/explore/file.dart -> depth 3.
        // We want to go back `depth - 1` times to reach `lib`.
        const prefix = depth > 1 ? '../'.repeat(depth - 1) : '';
        
        if (newContent.includes('AppRadii') && !newContent.includes('app_radii.dart')) {
            const importStmt = `import '${prefix}core/theme/app_radii.dart';\n`;
            newContent = newContent.replace(/(import .*;\n)/, `$1${importStmt}`);
        }
        if (newContent.includes('AppColors') && !newContent.includes('app_colors.dart')) {
            const importStmt = `import '${prefix}core/theme/app_colors.dart';\n`;
            newContent = newContent.replace(/(import .*;\n)/, `$1${importStmt}`);
        }
        
        fs.writeFileSync(filepath, newContent, 'utf8');
        console.log(`Updated ${filepath}`);
    }
});

File: refactor.py
import os
import re

directory = 'lib/features'

replacements = [
    (r'const Color\(0xFFFFE0B2\)', 'AppColors.warningContainer'),
    (r'const Color\(0xFFECF5FE\)', 'AppColors.surfaceContainerLow'),
    (r'Color\(0x0D002349\)', 'AppColors.primaryContainer.withValues(alpha: 0.05)'),
    
    (r'Radius\.circular\(\s*4(?:\.0)?\s*\)', 'AppRadii.sm'),
    (r'Radius\.circular\(\s*8(?:\.0)?\s*\)', 'AppRadii.defaultRadius'),
    (r'Radius\.circular\(\s*12(?:\.0)?\s*\)', 'AppRadii.md'),
    (r'Radius\.circular\(\s*16(?:\.0)?\s*\)', 'AppRadii.lg'),
    (r'Radius\.circular\(\s*24(?:\.0)?\s*\)', 'AppRadii.xl'),
    
    (r'BorderRadius\.circular\(\s*4(?:\.0)?\s*\)', 'AppRadii.borderSm'),
    (r'BorderRadius\.circular\(\s*8(?:\.0)?\s*\)', 'AppRadii.borderDefault'),
    (r'BorderRadius\.circular\(\s*12(?:\.0)?\s*\)', 'AppRadii.borderMd'),
    (r'BorderRadius\.circular\(\s*16(?:\.0)?\s*\)', 'AppRadii.borderLg'),
    (r'BorderRadius\.circular\(\s*24(?:\.0)?\s*\)', 'AppRadii.borderXl'),
    
    (r'(?<!Pdf)Colors\.red', 'AppColors.error'),
    (r'(?<!Pdf)Colors\.green', 'AppColors.success'),
    (r'(?<!Pdf)Colors\.amber', 'AppColors.warning'),
    (r'(?<!Pdf)Colors\.grey\.shade400', 'AppColors.outlineVariant'),
    (r'(?<!Pdf)Colors\.grey\.shade600', 'AppColors.outline'),
    (r'(?<!Pdf)Colors\.grey\.shade100', 'AppColors.surfaceContainerHigh'),
    (r'(?<!Pdf)Colors\.black26', 'AppColors.onSurface.withValues(alpha: 0.26)'),
    (r'(?<!Pdf)Colors\.black54', 'AppColors.onSurface.withValues(alpha: 0.54)'),
    (r'(?<!Pdf)Colors\.black\s*\.withValues\(\s*alpha:\s*0\.4\s*\)', 'AppColors.onSurface.withValues(alpha: 0.4)'),
    (r'(?<!Pdf)Colors\.black\s*\.withValues\(\s*alpha:\s*0\.15\s*\)', 'AppColors.onSurface.withValues(alpha: 0.15)'),
]

for root_dir, dirs, files in os.walk(directory):
    for file in files:
        if file.endswith('.dart'):
            filepath = os.path.join(root_dir, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
            
            new_content = content
            for pattern, repl in replacements:
                new_content = re.sub(pattern, repl, new_content)
                
            if new_content != content:
                # Calculate relative depth to lib/core
                depth = filepath.replace('\\', '/').count('/') - 1
                prefix = '../' * (depth - 1) if depth > 1 else ''
                
                if 'AppRadii' in new_content and 'app_radii.dart' not in new_content:
                    import_stmt = f"import '{prefix}core/theme/app_radii.dart';\n"
                    new_content = re.sub(r'(import .*;\n)', r'\1' + import_stmt, new_content, count=1)
                if 'AppColors' in new_content and 'app_colors.dart' not in new_content:
                    import_stmt = f"import '{prefix}core/theme/app_colors.dart';\n"
                    new_content = re.sub(r'(import .*;\n)', r'\1' + import_stmt, new_content, count=1)
                
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(new_content)
                print(f"Updated {filepath}")
